src/OrchardCore/OrchardCore.Users.Core/Authentication/CacheTicketStore.cs

```diff
@@ -0,0 +1,166 @@
+        var cacheKey = String.Concat(_keyPrefix, "-", key);
+
+        var userId = string.Empty;
+        var userName = string.Empty;
+
+        if (ticket.AuthenticationScheme == IdentityConstants.ApplicationScheme)
+        {
+            userId = ticket.Principal.FindFirst(ClaimTypes.NameIdentifier)?.Value; ;
+            userName = ticket.Principal.FindFirst(ClaimTypes.Name)?.Value ?? ticket.Principal.FindFirst("name")?.Value;
+        }
+
+        _dataProtector ??= _httpContextAccessor.HttpContext.RequestServices.GetService<IDataProtectionProvider>()
+                                .CreateProtector($"{nameof(CacheTicketStore)}_{IdentityConstants.ApplicationScheme}");
+
+        var protectedBytes = _dataProtector.Protect(SerializeTicket(ticket));
+        var data = new TicketRecord
+        {
+            Key = key,
+            UserId = userId,
+            Username = userName,
+            LastActivity = DateTime.UtcNow,
+            Value = protectedBytes,
+            Expires = ticket.Properties.ExpiresUtc.Value.UtcDateTime,
+        };
+
+        var cache = _httpContextAccessor.HttpContext.RequestServices.GetService<IDistributedCache>();
+        var ticketBytes = Serialize(data);
+        await cache.SetAsync(cacheKey, ticketBytes, new DistributedCacheEntryOptions() { AbsoluteExpiration = ticket.Properties.ExpiresUtc.Value });
+
+        return key;
+    }
+
+    private byte[] SerializeTicket(AuthenticationTicket source)
+        => TicketSerializer.Default.Serialize(source);
+
+    private AuthenticationTicket DeserializeTicket(byte[] source)
+        => source == null ? null : TicketSerializer.Default.Deserialize(source);
+
+    private byte[] Serialize(TicketRecord m)
+    {
+        using var ms = new MemoryStream();
+
+        using var writer = new BinaryWriter(ms);
+        writer.Write(m.Key);
+        writer.Write(m.UserId);
+        writer.Write(m.Username);
+        writer.Write7BitEncodedInt64(m.LastActivity.Ticks);
+        writer.Write7BitEncodedInt(m.Value.Length);
+        writer.Write(m.Value);
+        writer.Write7BitEncodedInt64(m.Expires.Value.Ticks);
+
+        return ms.ToArray();
+    }
+
+    private TicketRecord Deserialize(byte[] byteArray)
+    {
+        using var ms = new MemoryStream(byteArray);
+        using var reader = new BinaryReader(ms);
+        var ticket = new TicketRecord();
+        ticket.Key = reader.ReadString();
+        ticket.UserId = reader.ReadString();
+        ticket.Username = reader.ReadString();
+        var activityTicks = reader.Read7BitEncodedInt64();
+        ticket.LastActivity = new DateTime(activityTicks, DateTimeKind.Utc);
+        var bytesLength = reader.Read7BitEncodedInt();
+        ticket.Value = reader.ReadBytes(bytesLength);
+        var expireTicks = reader.Read7BitEncodedInt64();
+        ticket.Expires = new DateTime(expireTicks, DateTimeKind.Utc);
+
+        return ticket;
+    }
+}
+
+public record class TicketRecord
```

Comment:
Now I am confused about why you need this "enveloppe" since `TicketSerializer` can already do it all.

Reply:
In future - for List login/remote logout - i can just query cache and doesn't require to reserialize ticket

Reply:
That is where I think you are over confident. In the present you can't use these values because they are serialized as bytes, so useless. You could do that "in the future" too then, right now it's useless, and harmful (for perf).

Reply:
yes that' why I initially tried to serialize as json string

Reply:
Anyway for now, I'll remove that envelope - If "in the future" is performance concern then-  that I'll that UI in external module then.